sys-kernel/gentoo-kernel/gentoo-kernel-6.3.6-r1.ebuild

```diff
@@ -136,5 +136,81 @@ src_prepare() {
 		merge_configs+=( "${dist_conf_path}/big-endian.config" )
 	fi
 
+	if use modules-sign; then
+		: "${MODULES_SIGN_HASH:=sha512}"
+		cat <<-EOF > "${dist_conf_path}/modules-sign.config" || die
+			## Enable module signing
+			CONFIG_MODULE_SIG=y
+			CONFIG_MODULE_SIG_ALL=y
+			CONFIG_MODULE_SIG_FORCE=y
+			CONFIG_MODULE_SIG_${MODULES_SIGN_HASH^^}=y
+		EOF
+		if [ -n "${MODULES_SIGN_KEY}" ]; then
+			if [ -f "${MODULES_SIGN_KEY}" ]; then
+				echo "CONFIG_MODULE_SIG_KEY=\"${MODULES_SIGN_KEY}\"" \
+					>> "${dist_conf_path}/modules-sign.config"
+			else
+				die "MODULES_SIGN_KEY=${MODULES_SIGN_KEY} not found!"
+			fi
+		fi
+		merge_configs+=( "${dist_conf_path}/modules-sign.config" )
+	fi
+
 	kernel-build_merge_configs "${merge_configs[@]}"
 }
+
+src_install() {
+	kernel-build_src_install
+	if use modules-sign; then
+		local dir_ver=${PV}${KV_LOCALVERSION}
+		local kernel_dir=/usr/src/linux-${dir_ver}
+
+		# Copy the key/certificate so external modules can be signed
+		# If MODULES_SIGN_KEY is set no key was generated
+		if [ -z "${MODULES_SIGN_KEY}" ]; then
+			cp -p "build/certs/signing_key.pem" "${ED}${kernel_dir}/certs" || die
+			cp -p "build/certs/signing_key.x509" "${ED}${kernel_dir}/certs" || die
+		fi
```

Comment:
While convenient, the idea that the private key is going to be stored in binpkgs feels wrong, binpkgs are also readable by non-root by default.

Reply:
Is there any solution to this? If we don't install the key at all then we can't use `sign-file` to sign external modules.

Reply:
Probably the best way to avoid this problem is to encourage users to use an external key with `MODULES_SIGN_KEY`, this way the build system never generates the key and it is therefore never part of the binpkg.

I've added a warning message about this to `pkg_postinst`.